src/github.com/getlantern/balancer/balancer.go

```diff
@@ -71,13 +71,19 @@ func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error
 
 	_, port, _ := net.SplitHostPort(addr)
 
-	if port == "" || port == "80" || port == "8080" {
+	// try proxy non-https traffic through trusted servers first
+	if port != "443" {
```

Comment:
I reversed the test as http server may listen at arbitrary ports but https port is usually fixed.